frame/clearing-house/src/lib.rs

```diff
@@ -598,5 +901,84 @@ pub mod pallet {
 	impl<T: Config> Pallet<T> {}
 
 	// Helper functions - low-level functionality
-	impl<T: Config> Pallet<T> {}
+	impl<T: Config> Pallet<T> {
+		fn get_or_create_position<'a>(
+			positions: &'a mut BoundedVec<Position<T>, T::MaxPositions>,
+			market_id: &T::MarketId,
+			market: &Market<T>,
+		) -> Result<(&'a mut Position<T>, usize), DispatchError> {
+			Ok(match positions.iter().position(|p| p.market_id == *market_id) {
+				Some(index) =>
+					(positions.get_mut(index).expect("Item succesfully found above"), index),
+				None => {
+					positions
+						.try_push(Position::<T> {
+							market_id: market_id.clone(),
+							base_asset_amount: Zero::zero(),
+							quote_asset_notional_amount: Zero::zero(),
+							last_cum_funding: market.cum_funding_rate,
+						})
+						.map_err(|_| Error::<T>::MaxPositionsExceeded)?;
+					let index = positions.len() - 1;
+					let position = positions
+						.get_mut(index)
+						.expect("Will always succeed if the above push does");
+					(position, index)
+				},
```

Comment:
Even tough we know that all calls to `get_mut` here should succeed, I still think it's a good idea to not panic in case of any errors.
I'd recommend changing `.expect` to `.ok_or` variations, just in case.
(curious to know what you think tho :)

Reply:
I'm not sure. Let's raise this in next office hours to get the opinion of more people